### PR TITLE
Bug 1690681: Generate absolute URLs in emails

### PR DESCRIPTION
**Pull request by Rob--W**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1690681#c2

I haven't tested this patch, but guess that this may work based on reading code across the https://github.com/mozilla-services/phabricator-extensions, https://github.com/mozilla-conduit/phabricator and https://github.com/mozilla-conduit/phabricator-emails repositories.

### CODE REVIEW

**Comment by Rob--W** [comment]
@mitchhentges Could you review this? I haven't tested it but it should work.

**Comment by mitchhentges** [comment]
Tested this locally:
* Revision and user links are generated beautifully
* Project links (e.g.: `#bmo-core-security`) _still_ have relative links (`/tag/bmo-core/security`). This is because we tell Phabricator that we want "regularly rendered comments", not "HTML-email rendered comments", and the regularly-rendered-comments don't respect `uri.full` when handling projects.
    * However, if we switch to have Phabricator generate "HTML-email rendered comments" then we run into other issues, like Phab injecting its own CSS that uses `px` font sizing rather than relative font sizing. IIRC, there was other reasons why we didn't tell Phab to generate comments specifically for email, but I can't remember what those reasons were :disappointed:. Either way, this is something that we can doctor with a patch to core Phab code. But, since we hardly refer to projects, this isn't a large concern.

To summarize: nice investigation, patch looks good, let's land it once it's targeting `mozilla-conduit/phabricator` instead of `mozilla-services/phabricator-extensions`.
